### PR TITLE
Add insulin and physicalActivity types

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-321

### CODE REVIEW

File: TidepoolKit Example/Support/Extensions/Sample.swift [hunk 1]
```diff
@@ -47,6 +47,26 @@ struct Sample {
                            mealOther: "midnight snack",
                            nutrition: foodNutrition,
                            ingredients: [foodIngredient, foodIngredient]),
+                TInsulinDatum(time: Date(),
+                              dose: TInsulinDatum.Dose(active: 1.23,
+                                                       correction: 2.34,
+                                                       food: 3.45,
+                                                       total: 4.56),
```
Comment (nhamming) [inline TidepoolKit Example/Support/Extensions/Sample.swift:54]: [nit] I get that these are just arbitrary numbers, but should `total` be the sum of those above? Or is `total` something else?

File: TidepoolKit/Model/Datum/TInsulinDatum.swift [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+//  Created by Darin Krauss on 11/1/19.
+//  Copyright © 2019 Tidepool Project. All rights reserved.
+//
+
+import Foundation
+
+public class TInsulinDatum: TDatum, Decodable {
+    public var dose: Dose?
+    public var formulation: Formulation?
+    public var site: String?
```
Comment (nhamming) [inline TidepoolKit/Model/Datum/TInsulinDatum.swift:14]: Can you really have just a formulation or site without a dose?

File: TidepoolKit/Model/Datum/TPhysicalActivityDatum.swift [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        case preparationAndRecovery
+        case racquetball
+        case rowing
+        case rugby
+        case running
+        case sailing
+        case skatingSports
+        case snowboarding
+        case snowSports
+        case soccer
```
Comment (nhamming) [inline TidepoolKit/Model/Datum/TPhysicalActivityDatum.swift:66]: [nit] Interesting that there is `americanFootball` and `autralianFootball`, but only `soccer` and not `football`.

File: TidepoolKit/Model/Datum/TPhysicalActivityDatum.swift [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    }
+
+    public var name: String?
+    public var aggregate: Bool?
+    public var activityType: ActivityType?
+    public var activityTypeOther: String?
+    public var duration: Duration?
+    public var distance: Distance?
+    public var elevationChange: ElevationChange?
+    public var flight: Flight?
```
Comment (nhamming) [inline TidepoolKit/Model/Datum/TPhysicalActivityDatum.swift:103]: [nit] I personally find `flight` confusing and don't really know what I would put for a value.

**Review by nhamming — APPROVED** [review]
One comment and rest nits. Otherwise LGTM

**Review by ps2 — APPROVED** [review]
LGTM, in terms of style and structure. I haven't reviewed for conformance to the Tidepool API.

File: TidepoolKit Example/Support/Extensions/Sample.swift (hunk 1 above)
Comment (darinkrauss) [reply TidepoolKit Example/Support/Extensions/Sample.swift:54]: Total is how much was delivered. You can have some currently `active`, some *recommended* for `correction` to address a BG above target, some *recommended* for the `food` you just ate, and still dose a `total` different than the calculated `correction + food - active`.

File: TidepoolKit/Model/Datum/TInsulinDatum.swift (hunk 2 above)
Comment (darinkrauss) [reply TidepoolKit/Model/Datum/TInsulinDatum.swift:14]: No, you should not generally *create* a `TInsulinDatum` without a `dose`, which is why the initializer requires the `dose`. However, when parsing pre-existing data from the server, it is possible, though highly unlikely, to not have a dose.

File: TidepoolKit/Model/Datum/TPhysicalActivityDatum.swift (hunk 3 above)
Comment (darinkrauss) [reply TidepoolKit/Model/Datum/TPhysicalActivityDatum.swift:66]: When the backed data model for physical activity was created, I used the activities as found in HealthKit.

File: TidepoolKit/Model/Datum/TPhysicalActivityDatum.swift (hunk 4 above)
Comment (darinkrauss) [reply TidepoolKit/Model/Datum/TPhysicalActivityDatum.swift:103]: Again, this came from HealthKit, though I see it is called "totalFlightsClimbed".  This `flight` properties is a `Flight` struct which just contains a `count` int.

Comment (darinkrauss) [reply TidepoolKit/Model/Datum/TPhysicalActivityDatum.swift:103]: Perhaps not ideal, but it is what is published and public.

**Pull request merged**